### PR TITLE
Support vars file

### PR DESCRIPTION
**Pull request by till**
For variable substitution in `manifest.yml`.

Thoughts?

### CODE REVIEW

**Comment by till** [comment]
@iynere I was "assuming" that my `<<# parameters.vars>>` wraps this and omits it?

**Comment by iynere** [comment]
@till yes, that syntax is fine, i mean add a default value for the parameter where it is defined

File: src/@orb.yml [hunk 1]
```diff
@@ -123,6 +123,9 @@ jobs:
       manifest:
         type: string
         description: The Cloud Foundry manifest for this environment
+      vars:
+        type: string
```
Comment (iynere) [inline src/@orb.yml]: please add a default value here

File: src/@orb.yml [hunk 2]
```diff
@@ -181,6 +185,9 @@ jobs:
...
```
Comment (iynere) [inline src/@orb.yml]: please add a default value here

File: src/@orb.yml [hunk 3]
```diff
@@ -257,6 +265,9 @@ jobs:
...
```
Comment (iynere) [inline src/@orb.yml]: please add a default value here

File: src/@orb.yml [hunk 4]
```diff
@@ -372,6 +384,9 @@ commands:
...
```
Comment (iynere) [inline src/@orb.yml]: please add a default value here

File: src/@orb.yml [hunk 5]
```diff
@@ -399,6 +414,9 @@ commands:
...
```
Comment (iynere) [inline src/@orb.yml]: please add a default value here

File: src/@orb.yml (hunk 1 above)
Comment (till) [reply src/@orb.yml]: @iynere Ah, I figured it would work the same as the `manifest.yml`? Which default would you like, empty string?

Comment (till) [reply src/@orb.yml]: Or `package`.

**Comment by iynere** [comment]
yes, empty string is probably best, thank you ! @till 

i'm not super familiar w/ cloudfoundry—are there situations where folks would run these commands / jobs without using a manifest file, as well ? if so, would you mind adding defaults for that parameter, as well ? (empty string is good in that case, as well.)

thanks so much !

**Comment by till** [comment]
@iynere the `vars-file` is only supposed to be used with a manifest.

Think of it like this:

```
---
applications:
- name: foo
  memory: ((memory)
  path: .
  buildpacks:
  - nodejs_buildpack
```

Then add multiple variable files:

```
# vars-dev.yml
memory: 256M
```

```
# vars-prod.yml
memory: 2G
```

Then deploy with `cf push ... --vars-file vars-prod.yml`. The manifest default is `./manifest.yml`. You don't need to specify that. If you don't have a `manifest.yml` to begin with, then you don't need a vars file to substitute variables.

You can also have multiple manifests and then specify one using the parameter — or if the default name doesn't fit your bill.

Can you let me know know:

 1. Do you still want a default value for the `manifest`?
 2. How do jobs currently fail without an empty string for `vars`? 

I am not entirely sure I understand.

**Comment by iynere** [comment]
@till if a parameter does not have a default value it is considered a 'required' parameter, which means a circleci job will not run at all if the config does not pass a value for that parameter

@till yes, please add a default value (`""`) for both the `manifest` and `vars` parameters, thank you!

**Comment by till** [comment]
@iynere Thanks for taking the time to review/explain. I hope this works now!

**Comment by iynere** [comment]
looks 👌 to me, ty, appreciate it !! @till

**Comment by iynere** [comment]
hmm, PR is failing & i'm not sure why, @eddiewebb wrote these tests so perhaps he can help debug

**Comment by till** [comment]
@iynere Where's the build?

**Comment by iynere** [comment]
@till https://circleci.com/gh/CircleCI-Public/cloudfoundry-orb/50 if you want to help debug—thank you !

**Comment by eddiewebb** [comment]
@iynere / @till  - the additional parameter is failing the test assertin because it was not expected.

We want a test case that covers empty and non-empty var file parameter, and how the generated output should vary.

https://github.com/CircleCI-Public/cloudfoundry-orb/blob/master/test/test_expansion.bats#L98

**Comment by till** [comment]
https://github.com/CircleCI-Public/cloudfoundry-orb/pull/14

**Comment by till** [comment]
I think "contains" should not break it, as the output didn't change.

We should probably add another test!

**Pull request merged**